src/OpenTelemetry/Metrics/Aggregators/Aggregator.cs

```diff
@@ -28,31 +28,44 @@ namespace OpenTelemetry.Metrics.Aggregators
     public abstract class Aggregator<T>
         where T : struct
     {
-        private long startTimeTicks;
-        private long checkpointStartTimeTicks;
+        private AggState active;
+        private AggState checkpoint;
 
         protected Aggregator()
         {
-            this.startTimeTicks = DateTimeOffset.UtcNow.Ticks;
+            this.checkpoint = new AggState();
+            this.active = new AggState();
         }
 
         /// <summary>
         /// Adds value to the running total in a thread safe manner.
         /// </summary>
         /// <param name="value">Value to be aggregated.</param>
-        public abstract void Update(T value);
+        public virtual void Update(T value)
+        {
+            this.active.Increment();
+        }
 
         /// <summary>
         /// Checkpoints the current aggregate data, and resets the state.
         /// </summary>
         public virtual void Checkpoint()
         {
             // checkpoints the start time for the current aggregation, and sets the new start time.
-            this.checkpointStartTimeTicks = Interlocked.Exchange(ref this.startTimeTicks, DateTimeOffset.UtcNow.Ticks);
+            this.checkpoint = Interlocked.Exchange(ref this.active, new AggState());
+        }
+
+        /// <summary>
+        /// Check if checkpoint has any aggregated data.
+        /// </summary>
+        /// <returns>true if data was presented to aggregator.</returns>
+        public virtual bool HasData()
```

Comment:
Couple of questions...
* Any reason you went with method over property (`public virtual bool HasData => this.checkpoint.Count > 0;`)?
* Should we call it `HasCheckpointData`? Just looking at the code it looks like there's actually 2 counts in play (`active` & `checkpoint`).

Reply:
I try to follow the style in existing code.  In this section of code, I do not see much use of Properties.  I am not that opinionated on this topic, so if you prefer, I can change to property.

Reply:
I don't feel strongly about it either, was mostly just curious 😄